kb/development/test_suites.md

```diff
@@ -196,3 +196,30 @@ In a test case that always uses PSA crypto, call `PSA_INIT()` at the beginning a
 In a test case that uses PSA crypto only when building with `MBEDTLS_USE_PSA_CRYPTO`, call `USE_PSA_INIT()` at the beginning and `USE_PSA_DONE()` at the end.
 
 See [`tests/include/test/psa_crypto_helpers.h`](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/include/test/macros.h) for more complex cases.
+
+## Guidance on writing unit test data
+
+### Document the test data
+
+Document how the test data was generated. This helps future maintainers if they need to generate more similar test data, for example to construct a non-regression test for a bug in a particular case.
+
+The documentation can be:
+
+* In Python (or other) code committed to the repository. This is preferred when the code can handle a large class of tests. For example, we have frameworks to generate key data for PSA, and to generate bignum tests. Some `.data` files are fully generated at build time by `tests/scripts/generate_*_tests.py`.
+* In a comment in the `.data` file. This is more convenient when the instructions are specific to a single test case.
+* In the commit that introduces the test data. This is more convenient when the instructions cover a series of similar test cases.
+
+When adding a new file in `tests/data_files`, if possible, do it by editing `tests/data_files/Makefile` to add executable instructions for creating those files, and then run `make` and commit the result. Ideally removing the files and re-running `make` should produce identical files, however in many cases this is not practical because the generation is randomized or depends on the current time (for certificates and related data), and we accept that.
+
+### Interoperability testing
+
+Test data for cryptographic algorithms should be, in order of preference:
+
+* Official test vectors taken from a standards document.
+* Generated by a reference implementation from the authors of the algorithm.
+* Generated by a widespread implementation such as OpenSSL or Cryptodome.
```

Comment:
On second thoughts I don't think we should systematically privilege implementations from the authors. Yes, they're more official, but they're also often harder to use. So I don't want to discourage using a popular implementation like OpenSSL. Obviously if we suspect an OpenSSL bug we can dig down further — and even so we might privilege compatibility with the de facto standard over the de jure standard.